modules/ROOT/pages/http/http-listener-ref.adoc

```diff
@@ -1,182 +1,296 @@
+It's important to note that while a reason phrase is not defined for error responses,
+ the connector will attempt to define it based on the status code. Thus, a `Gateway Timeout` or `Internal Server Error` will be returned in the scenarios explained before.
 
-== Generating the HTTP Response
-
-After request processing, the message is returned to the HTTP connector to provide a response. You can customize the response body, attachment, headers, and status.
-
-=== HTTP Response Body
-
-The HTTP Response body is generated from payload of the Mule Message after executing the flow unless explicitly configured. You can use a DataWeave script or expression to customise it.
-
-=== HTTP Response Headers
+[source,xml,linenums]
+----
+<flow name="store-logo">
+  <http:listener config-ref="HTTP_Listener_config" path="/account/{accountId}/logo">
+    <http:response statusCode="201" reasonPhrase="Created">
+      <http:body ><![CDATA[#[output text/plain --- vars.logoName ++ ' has been stored as a ' ++ vars.accountId ++ ' logo.']]]></http:body>
+    </http:response>
+    <http:error-response statusCode="#[vars.errorCode]">
+      <http:body ><![CDATA[#[vars.logoName ++ ' could not be stored.']]]></http:body>
+      <http:headers ><![CDATA[#[
+        output application/java
+        ---
+        {
+          "X-Time" : "50s"
+        }
+      ]]]></http:headers>
+    </http:error-response>
+  </http:listener>
+  <set-variable variableName="logoName" value="#[payload.parts.name.content]" mimeType="text/plain"/>
+  <set-variable variableName="accountId" value="#[attributes.uriParams.accountId]" mimeType="text/plain"/>
+  <!-- store logo -->
+  <error-handler >
+    <on-error-propagate type="CONNECTIVITY">
+      <set-variable variableName="errorCode" value="504"/>
+    </on-error-propagate>
+  </error-handler>
+</flow>
+----
 
-The response message sent by the HTTP Listener can include HTTP headers. In Headers, enter an expression to include information in the response, for example: `#[{'Connection' : 'close' }]`. This will add a `Connection` header with `close` value to the HTTP response.
+==== HTTP Streaming Mode
 
-=== HTTP Response Status Code and Reason Phrase
+When handling response bodies, the HTTP connector will consider the kind of data to send and use `chunked` encoding when the size is not clear (think streams with no size information). You can change this behavior using the `responseStreamingMode` options:
 
-You can customize the connector response to requests dynamically. For example, you can set different error status code numbers depending on the error raised by having an error handler that sets the value of a variable used to define the HTTP response status code.
+* AUTO (default): if a size is defined for the body it will use `Content-Length` encoding, otherwise it will use `Transfer-Encoding: chunked`.
```

Comment:
"body it" -> "body, the connector" (safer/clearer to say what "it" refers to)

"encoding, otherwise" -> "encoding. Otherwise, "